tests/unit/utils/test_logger.py

```
@@ -36,6 +41,33 @@ def test_passes_through_safe_text(self):
+            },
+            "credentials": {
+                "client_secret": "oauth-secret",
+                "password": "database-password",
+            },
+        }
+
+        filtered = filter_sensitive_value(value)
+
+        assert filtered["headers"]["Authorization"] == "***REDACTED***"
+        assert filtered["headers"]["X-Request-ID"] == "request-123"
+        assert filtered["credentials"]["client_secret"] == "***REDACTED***"
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to: "client_secret"

(S105)